### PR TITLE
feat: custodian s3 RFC

### PR DESCRIPTION
**Pull request by olevski**
This is an outline of how we can use the Custodian to inject s3 credentials in requests for launching notebook sessions for the specific project we discussed.

Please feel free to comment and to commit to this branch.

### CODE REVIEW

**Comment by rokroskar** [comment]
@martinfontanet could you please weigh in here on the RFC? If it looks good and is workable we should finalize it and merge.

**Comment by martinfontanet** [comment]
I just added some changes to the diagram in #10.

**Regarding the first question**:
At the moment, the Custodian uses the Renku's Keycloak instance as IDP. It uses OpenID connect (from Renku Keycloak) to authenticate users at both the contract signature step and the session creation step, so it does not need any extra credentials. This means that the link only requires the necessary parameters to start a session with the Renku API (except the S3 credentials, which are handled by the Custodian).

This also partly answers the third question, as there are currently no users that are registered with the Custodian but not with Renku.

**Comment by rokroskar** [comment]
@olevski @martinfontanet what's the status here? would be nice to merge it if it's being used/implemented/considered, otherwise lets close it?

**Comment by olevski** [comment]
@rokroskar @martinfontanet has done some work on this as far as I know. Not sure where the code is or what the status is.

**Comment by olevski** [comment]
@rokroskar this is now up-to-date and ready to be merged.

**Review by martinfontanet — APPROVED** [review]

File: design/005-custodian-s3-credentials/005-custodian-s3-credentials.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+- Start Date: 2022-04-28
+- Status: Draft
```
Comment (rokroskar) [inline design/005-custodian-s3-credentials/005-custodian-s3-credentials.md:2]: I presume this is now beyond a draft? e.g. "partially implemented"?

**Review by rokroskar — CHANGES_REQUESTED** [review]
Please update the status

File: design/005-custodian-s3-credentials/005-custodian-s3-credentials.md (hunk 1 above)
Comment (olevski) [reply design/005-custodian-s3-credentials/005-custodian-s3-credentials.md:2]: Yeah I will change that

**Review by rokroskar — APPROVED** [review]

**Pull request merged**